doc/examples/optional_headers.rst

```diff
@@ -0,0 +1,68 @@
+Optional Headers
+^^^^^^^^^^^^^^^^
+
```

Comment:
I think it would be good to add a brief reasoning for including these or why they are potentially useful